pkg/provider/pulumi/common/build.go

```diff
@@ -29,7 +29,7 @@ import (
 
 func dynamicDockerfile(dir, name string) (*os.File, error) {
 	// create a more stable file name for the hashing
-	return os.CreateTemp(dir, "nitric."+name+".Dockerfile.*")
+	return os.Create(filepath.Join(dir, "nitric."+name+".Dockerfile"))
```

Comment:
this is one side-effect, it requires that the docker file is in a subdir of the context.. so we can't put it in a temp dir :shrug:

Reply:
that's interesting, they are just executing `buildx` using `-f` for the file reference https://github.com/MaterializeInc/pulumi-docker-buildkit/blob/90a12e5f63871beb10ad1e4ffc6b69abb735281b/cmd/pulumi-resource-docker-buildkit/main.go#L268-L288. 

Last time I tested this with buildkit referencing dockerfiles outside of build context worked fine. Might be worth trying to just run a built like this locally on your machine (e.g. using the CLI image but move the dockerfile somewhere outside of context).

Just tested it locally then with the CLI, and the -f reference works outside of context both relative and absolute.

Reply:
the issue is here:
https://github.com/MaterializeInc/pulumi-docker-buildkit/blob/90a12e5f63871beb10ad1e4ffc6b69abb735281b/cmd/pulumi-resource-docker-buildkit/main.go#L240
https://github.com/MaterializeInc/pulumi-docker-buildkit/blob/90a12e5f63871beb10ad1e4ffc6b69abb735281b/cmd/pulumi-resource-docker-buildkit/main.go#L419
https://github.com/MaterializeInc/pulumi-docker-buildkit/blob/90a12e5f63871beb10ad1e4ffc6b69abb735281b/cmd/pulumi-resource-docker-buildkit/main.go#L373

it joins the contextDir and the dockerfile path